padle/padle.py

```diff
@@ -0,0 +1,537 @@
+import discord
+import datetime
+import random
+import asyncio
+import logging
+
+from typing import List, Literal, Optional
+from redbot.core import Config, commands
+from math import ceil
+
+# from dbcog.monster_graph import MonsterGraph # used for creating the filter
+from dbcog.dbcog import DBCog
+from tsutils.tsubaki.custom_emoji import get_awakening_emoji, get_rarity_emoji, get_type_emoji, get_emoji
+from tsutils.tsubaki.links import MonsterImage, MonsterLink
+from tsutils.tsubaki.monster_header import MonsterHeader
+
+from tsutils.cogs.userpreferences import get_user_preference
+from tsutils.helper_functions import conditional_iterator
+from tsutils.user_interaction import get_user_confirmation, send_confirmation_message, send_cancellation_message
+from tsutils.emoji import NO_EMOJI, SendableEmoji, YES_EMOJI
+
+
+from discordmenu.embed.components import EmbedThumbnail, EmbedMain
+from discordmenu.embed.view import EmbedView
+from redbot.core.utils import menus
+from redbot.core.utils.menus import (
+    next_page,
+    prev_page,
+    start_adding_reactions
+)
+
+logger = logging.getLogger('red.padbot-cogs.padle')
+
+class PADle(commands.Cog):
+    """PADle"""
+
+    def __init__(self, bot, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.bot = bot
+        self.config = Config.get_conf(self, identifier=630817360)
+        self.config.register_user(guesses=[], start=False, done=False, score=[],
+                                  edit_id="", channel_id="", embed_text=[])
+        self.config.register_global(padle_today=3260, stored_day=0, num_days=0,
+                                    subs=[], all_scores=[], save_daily_scores=[])
+
+        self._daily_padle_loop = bot.loop.create_task(self.generate_padle())
+
+    async def get_dbcog(self) -> "DBCog":
+        dbcog = self.bot.get_cog("DBCog")
+        if dbcog is None:
+            raise ValueError("DBCog cog is not loaded")
+        await dbcog.wait_until_ready()
+        return dbcog
+
+    @commands.group()
+    async def padle(self, ctx):
+        """Commands pertaining to PADle"""
+
+    @padle.command()
+    async def help(self, ctx):
+        """Instructions for PADle"""
+        prefix = (await self.bot.get_prefix(ctx.message))[0]
```

Comment:
Red v3 has the nice and easy `ctx.prefix` which is way easier than the asynchronous `bot.get_prefix(message)` stuff that some cogs still use.  It's also more reliable for bots with multiple prefixes.